### PR TITLE
Vue training - 06 - Internationalization

### PR DESCRIPTION
**Pull request by rsuescun**
## Summary

-Added translations logic

## Trello Card

https://trello.com/c/lxKZlhhg/19-internationalization

### CODE REVIEW

File: vue-training/src/views/index/index.js [hunk 1]
```diff
@@ -1,23 +1,38 @@
 import Vue from 'vue'
+import VueI18n from 'vue-i18n'
+
+Vue.use(VueI18n)
 
 import { installServiceWorker } from '../../serviceWorkerInstaller'
 
+import { DEFAULT_LANGUAGE } from '../../constants/general.js'
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.js]: remove file extension from imports

File: vue-training/src/views/index/index.js [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+
 import TechService from '../../services/techsService.js'
 
+import TRANSLATIONS from '../../translations/i18n.js'
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.js]: same

Comment (FrankIglesias) [reply vue-training/src/views/index/index.js]: and I think you can use alias

File: vue-training/src/views/index/index.pug [hunk 3]
```diff
@@ -9,16 +9,24 @@ html(lang='en')
     link(rel='preload' href='styles.index.css' as='style')
     link(rel='stylesheet' href='styles.index.css')
     script(src='./vendor.js')
+    script(src='https://unpkg.com/vue/dist/vue.js')
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.pug]: Do you nee this? this is set inside the  webpack  file already

File: vue-training/src/views/index/index.pug [hunk 4]
```diff
@@ -9,16 +9,24 @@ html(lang='en')
...
+    script(src='https://unpkg.com/vue-i18n/dist/vue-i18n.js')
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.pug]: use npm

File: vue-training/src/views/index/index.pug [hunk 5]
```diff
@@ -9,16 +9,24 @@ html(lang='en')
...
-          span.filter-count
-            | {{ countTechs }} Tecnología(s)
-          input.filter-input(type="text" placeholder="Filtrar" v-model="filter")
+          img.navbar-logo(
+            src='/assets/wolox_logo.svg'
+            :alt="$t('navbar.altLogo')"
+          )
+          h1.navbar-text {{ $t("navbar.title") }}
+        .navbar-bottom-container
+          span.filter-count {{ $tc('navbar.filter.countMessage', countTechs, { count: countTechs }) }}
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.pug]: pipe here

File: vue-training/webpack.config.js [hunk 6]
```diff
@@ -45,6 +45,11 @@ module.exports = {
         test: /\.vue$/,
         loader: 'vue-loader'
       },
+      {
```
Comment (FrankIglesias) [inline vue-training/webpack.config.js]: why this?

File: vue-training/techs.json [hunk 7]
```diff
@@ -1,68 +1,136 @@
...
-          "title": "Django",
-          "launch": "2005",
-          "description": "Django (/ˈdʒæŋɡoʊ/ JANG-goh)[5] is a free and open-source web framework, written in Python, which follows the model-view-template (MVT) architectural pattern.",
-          "logo": "/assets/tech/django_icon.svg"
-        }
-      ]
-    }
-  ]
-}
+  "techs": {
```
Comment (FrankIglesias) [inline vue-training/techs.json]: use a js instead

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: vue-training/src/views/index/index.pug (hunk 3 above)
Comment (rsuescun) [reply vue-training/src/views/index/index.pug]: I forgot to remove it

File: vue-training/techs.json [hunk 8]
```diff
@@ -1,68 +1,136 @@
...
+            "title": "Django",
+            "launch": "2005",
+            "description": "Django (/ˈdʒæŋɡoʊ/ JANG-goh)[5] es un marco web gratuito y de código abierto, escrito en Python, que sigue el patrón arquitectónico modelo-vista-plantilla (MVT).",
+            "logo": "/assets/tech/django_icon.svg"
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (FrankIglesias) [inline vue-training/techs.json]: add EOL

File: vue-training/webpack.config.prod.js [hunk 9]
```diff
@@ -39,6 +39,11 @@ module.exports = {
         test: /\.vue$/,
         use: 'vue-loader'
       },
+      {
```
Comment (FrankIglesias) [inline vue-training/webpack.config.prod.js]: Why this webpack conf?

File: vue-training/webpack.config.js [hunk 10]
```diff
@@ -45,6 +45,11 @@ module.exports = {
...
+        resourceQuery: /blockType=i18n/,
```
Comment (FrankIglesias) [inline vue-training/webpack.config.js]: Same

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: vue-training/src/components/TechBox.vue [hunk 11]
```diff
@@ -1,11 +1,13 @@
 <template lang="pug">
   .tech-box
+    //- TODO: resolve error when I add this :alt='tech.title'
```
Comment (FrankIglesias) [inline vue-training/src/components/TechBox.vue]: Why are you having this issue?

File: vue-training/src/translations/i18n.js [hunk 12]
```diff
@@ -0,0 +1,52 @@
+const TRANSLATIONS = {
```
Comment (FrankIglesias) [inline vue-training/src/translations/i18n.js]: why not export const directly

File: vue-training/techs.js [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+          launch: '2005',
+          description: 'Django (/ˈdʒæŋɡoʊ/ JANG-goh)[5] es un marco web gratuito y de código abierto, escrito en Python, que sigue el patrón arquitectónico modelo-vista-plantilla (MVT).',
+          logo: '/assets/tech/django_icon.svg'
+        }
+      ]
+    }
+  ]
+}
+
+module.exports = () => ({ techs })
```
Comment (FrankIglesias) [inline vue-training/techs.js:136]: why this?

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: vue-training/src/components/TechBox.vue (hunk 11 above)
Comment (rsuescun) [reply vue-training/src/components/TechBox.vue]: It works but shows this error when compiling 
Screen Shot 2019-04-29 at 11 01 10 AM

File: vue-training/techs.js (hunk 13 above)
Comment (rsuescun) [reply vue-training/techs.js:136]: How do I return a json with `export default`?

File: vue-training/src/components/TechBox.vue (hunk 11 above)
Comment (FrankIglesias) [reply vue-training/src/components/TechBox.vue]: missing ,?

File: vue-training/techs.js (hunk 13 above)
Comment (FrankIglesias) [reply vue-training/techs.js:136]: it's an object, you can just do `export const {....`

File: vue-training/src/components/TechBox.vue (hunk 11 above)
Comment (rsuescun) [reply vue-training/src/components/TechBox.vue]: yes, that was solved

File: vue-training/techs.js (hunk 13 above)
Comment (rsuescun) [reply vue-training/techs.js:136]: I do not know what I'm doing, but that does not work for me

Comment (rsuescun) [reply vue-training/techs.js:136]: In any case this point does not seem very important to me because it's just a api mock

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**